### PR TITLE
Replace sigatomic_(begin|end) with disable_sigint() do ... end

### PR DESCRIPTION
**Pull request by mrufsvold**
Based on [this issue](https://github.com/JuliaIO/VideoIO.jl/issues/371), I switched out all of the uses of sigatomic with disable_sigint. Because the functions are now wrapped in a do block, I had to define `ret` as a `Ref{Int32}` outside the block to make it accessible to the error catches. 

I tested `load()`, `get_number_frames()`, and `get_duration()` both in the REPL and using the VSCode debugger. They all worked in both cases now.

I'm not sure if there are other tests that should be run, but hopefully, someone more knowledgeable about this package and verify all is well!

### CODE REVIEW

File: src/avio.jl [hunk 1]
```diff
@@ -116,10 +116,13 @@ convert(::Type{AVRational}, r::Rational) = AVRational(numerator(r), denominator(
...
-        ret = av_read_frame(avin.format_context, avin.packet)
-        sigatomic_end()
-        if ret < 0
+        # Need to declare ret outside the do block
+        ret = Ref{Int32}()
+        disable_sigint() do
+            ret[] = av_read_frame(avin.format_context, avin.packet)
+        end
+
+        if ret[] < 0
```
Comment (IanButterworth) [inline src/avio.jl:125]:
```suggestion
        ret = disable_sigint() do
            av_read_frame(avin.format_context, avin.packet)
        end
        if ret < 0
```

File: src/avio.jl [hunk 2]
```diff
@@ -296,12 +299,13 @@ function VideoReader(
...
-    unlock(VIO_LOCK)
-    sigatomic_end()
-    ret < 0 && error("Could not open codec")
+    ret = Ref{Int32}()
+    disable_sigint() do
+        lock(VIO_LOCK)
+        ret[] = avcodec_open2(codec_context, codec, C_NULL)
+        unlock(VIO_LOCK)
+    end
+    ret[] < 0 && error("Could not open codec")
```
Comment (IanButterworth) [inline src/avio.jl:308]:
```suggestion
    ret = disable_sigint() do
        lock(VIO_LOCK) do
            avcodec_open2(codec_context, codec, C_NULL)
        end
    end
    ret < 0 && error("Could not open codec")
```

File: src/encoding.jl [hunk 3]
```diff
@@ -324,12 +324,13 @@ options, or pass the private options to `encoder_private_options` explicitly""",
...
-    unlock(VIO_LOCK)
-    sigatomic_end()
-    ret < 0 && error("Could not open codec: Return code $(ret)")
+    ret = Ref{Int32}()
+    disable_sigint() do
+        lock(VIO_LOCK)
+        ret[] = avcodec_open2(codec_context, codec, C_NULL)
+        unlock(VIO_LOCK)
+    end
+    ret[] < 0 && error("Could not open codec: Return code $(ret)")
```
Comment (IanButterworth) [inline src/encoding.jl:333]:
```suggestion
    ret = disable_sigint() do
        lock(VIO_LOCK) do
            avcodec_open2(codec_context, codec, C_NULL)
        end
    end
    ret < 0 && error("Could not open codec: Return code $(ret)")
```

File: src/frame_graph.jl [hunk 4]
```diff
@@ -134,18 +134,19 @@ function exec!(s::SwsTransform)
...
+                s.sws_context,
+                src_data_ptr,
+                src_linesize_ptr,
+                0,
+                s.dstframe.height,
+                dst_data_ptr,
+                dst_linesize_ptr,
+            )
+        end
+        ret[] <= 0 && error("Could not scale frame")
```
Comment (IanButterworth) [inline src/frame_graph.jl:149]:
```suggestion
        ret = disable_sigint() do
            sws_scale(
                s.sws_context,
                src_data_ptr,
                src_linesize_ptr,
                0,
                s.dstframe.height,
                dst_data_ptr,
                dst_linesize_ptr,
            )
        end
        ret <= 0 && error("Could not scale frame")
```

File: .gitignore [hunk 5]
```diff
@@ -10,3 +10,4 @@ test/precisiontest_gray_test.mp4
 test/ordertest_gray_test.mp4
 docs/build/
 test/temp.stream
+.vscode/
```
Comment (IanButterworth) [inline .gitignore:13]:
```suggestion
.vscode/

```

**Review by IanButterworth — CHANGES_REQUESTED** [review]
Thanks for this.

I don't think the `Ref`'s are needed? Also the locks can be tidied up

**Comment by mrufsvold** [comment]
Addressed your feedback!

File: src/avio.jl [hunk 6]
```diff
@@ -116,9 +116,11 @@ convert(::Type{AVRational}, r::Rational) = AVRational(numerator(r), denominator(
 # Pump input for data
 function pump(avin::AVInput)
     while avin.isopen && !avin.finished
-        sigatomic_begin()
-        ret = av_read_frame(avin.format_context, avin.packet)
-        sigatomic_end()
+        # Need to declare ret outside the do block
```
Comment (IanButterworth) [inline src/avio.jl:119]:
```suggestion
```

**Review by IanButterworth — APPROVED** [review]
Thanks. Sorry for the delay

**Comment by IanButterworth** [comment]
There seems to be a problem with nightly. Adding 1.8-nightly to evaluate.

I'll merge either way and fix on another PR

**Pull request merged**